common/web/input-processor/src/text/prediction/languageProcessor.ts

```diff
@@ -125,20 +132,21 @@ export default class LanguageProcessor extends EventEmitter<LanguageProcessorEve
 
     let specType: 'file'|'raw' = model.path ? 'file' : 'raw';
     let source = specType == 'file' ? model.path : model.code;
-    let lp = this;
 
     // We pre-emptively emit so that the banner's DOM elements may update synchronously.
     // Prevents an ugly "flash of unstyled content" layout issue during keyboard load
     // on our mobile platforms when embedded.
-    lp.currentModel = model;
+    this.currentModel = model;
     if(this.mayPredict) {
-      lp.emit('statechange', 'active');
+      this._state = 'active';
+      this.emit('statechange', 'active');
```

Comment:
Feels like it would be good to avoid duplicating this `'active'` string, also a few similar duplications further down in this file.

```
emitStateChange(): void {
  this.emit('statechange', this._state);
}
```